### PR TITLE
Modify ncdump to print char-valued variables as utf8.

### LINKED ISSUE #2916: Request: ncdump: Fix printing of UTF-8 in "character" data variables
This is a feature request.  I tested this with netcdf-c 4.9.2 on mac and linux, but I think it applies to all previous netcdf versions.

Ncdump has this convenient feature.  When printing **type character**, it concatenates single chars along the right hand dimension, and prints them as strings.  This is nice for embedding actual string data in classic netcdf-3 format where there is no actual **"string" data type**.

However, ncdump does not correctly print UTF-8 strings when embedded like this in **type character**.  UTF-8 multi-byte characters in the data are split into single bytes, and printed as octal escape sequences.  See example printout below, and the netcdf-4 file for this demo is [attached here.](https://github.com/Unidata/netcdf-c/files/15195407/utf8.test.nc4.zip)

Can ncdump be updated to print such UTF-8 character strings correctly?  I notice that **"string" data type** in netcdf-4 is handled differently, and UTF-8 is rendered correctly in that case.
```
netcdf utf8.test {
dimensions:
	site = 4 ;
	len25 = 25 ;
variables:
	string name1(site) ;
	char name2(site, len25) ;
data:

 name1 = "Sainte-Anne-de-Bellevue  ",   // name1(0)
    "Aéroport de Montréal 1 ",   // name1(1)
    "Rivière-des-Prairies    ",   // name1(2)
    "Chenier                  ";  // name1(3)
    
 name2 =
  "Sainte-Anne-de-Bellevue  ",  // name2(0,24)
    "A\303\251roport de Montr\303\251al 1 ",  // name2(1,24)
    "Rivi\303\250re-des-Prairies    ",  // name2(2,24)
    "Chenier                  ";  // name2(3,24)
    }
```

### PR DESCRIPTION
**Pull request by DennisHeimbigner**
* Fixes #2916 

re: Issue https://github.com/Unidata/netcdf-c/issues/2916

Currently, ncdump prints char-valued variables as a mix
of ascii and octal characters. The octal format is used
for non-printable ascii character values.

This PR changes this to print the char variable values
as raw binary. This means in practice that utf-8 tags
are properly interpreted and printed as utf-8.

### CODE REVIEW

**Review by WardF — APPROVED** [review]

File: .github/workflows/run_tests_s3.yml
Comment (WardF) [inline .github/workflows/run_tests_s3.yml:1]: @DennisHeimbigner why was this file deleted?

Comment (DennisHeimbigner) [reply .github/workflows/run_tests_s3.yml:1]: It was never being used. Its purpose was to show in theory how to store the secret key
for accessing the Unidata S3 bucket. But it never worked so I decided to delete it.
We can return to it if/when we figure out how to do this safely.

Comment (DennisHeimbigner) [reply .github/workflows/run_tests_s3.yml:1]: An addendum.
The action can still test S3 read access if you set the following options:
````
--netcdf-enable-s3 --with-s3-testing=public
````

Comment (WardF) [reply .github/workflows/run_tests_s3.yml:1]: Thanks Dennis!
